pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -735,4 +743,249 @@ public class Utils {
         }
     }
 
+    /**
+     * Translate a list of objects which may either be instances of a given class or {@link UninstantiatedDescribable}s,
+     * and return a list of those instances of the class and instantiated version of those {@link UninstantiatedDescribable}s.
+     *
+     * @param clazz The class we'll be instantiating, which must implement {@link Describable}.
+     * @param toInstantiate The list of either instances of the class or {@link UninstantiatedDescribable}s that can be
+     * instantiated to instances of the class.
+     * @return The list of instances. May be empty.
+     */
+    @Nonnull
+    private static <T extends Describable> List<T> instantiateList(Class<T> clazz, List<Object> toInstantiate) {
+        List<T> l = []
+        toInstantiate.each { t ->
+            if (t instanceof UninstantiatedDescribable) {
+                l.add((T) t.instantiate())
+            } else if (clazz.isInstance(t)) {
```

Comment:
Good point. Originally I had this check ahead of the `UninstantiatedDescribable` check, so it made sense, but that didn't work at all for whatever reason, so I flipped them and didn't change this. Fixing!